packages/core/src/utils/memoryDiscovery.ts

```diff
@@ -234,20 +234,29 @@ async function readGeminiMdFiles(
   const results: GeminiFileContent[] = [];
   for (const filePath of filePaths) {
```

Comment:
We should rather make sure we never pass directories in filePaths in the first place.

Reply:
Sorry we missed this question. You are correct about where to make that change. After you make that change and add a test for this I'll approve.
Thanks again for this robustness polish!